notebooks/Liquidators_data.ipynb

```diff
@@ -0,0 +1,977 @@
+    "3. Identify the amounts of debt tokens liquidated.\n",
+    "4. Visualize the behavior of all liquidators found in the sample in a Jupyter notebook. You should use the visualizations to answer the following questions:\n",
+    "\n",
+    "How many liquidators are there per protocol? Do some of them operate on more than just a single lending protocol?\n",
+    "\n",
+    "How many loans has each of them liquidated?\n",
+    "\n",
+    "What is the average liquidated debt amount? Compute the averages across all liquidations, but also per protocol and per token amounts.\n",
+    "\n",
+    "What is the total amount of the liquidated debt? Compute the total amounts across all liquidations, but also per protocol and per token amounts.\n",
+    "\n",
+    "Fetch prices for these tokens (in USD), e.g., from Coingecko. Multiply the token amounts by the corresponding prices to obtain USD nominal amounts and be able to compare the per token stats across tokens."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 167,
+   "id": "bf107fa7",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Requirement already satisfied: psycopg2 in c:\\users\\molak\\anaconda3\\lib\\site-packages (2.9.9)\n",
+      "env: PG_CONNECTION_STRING=''  # nostra_mainnet.py has dependencies. To avoid error adding dummy env\n"
+     ]
+    }
+   ],
+   "source": [
+    "!pip install psycopg2\n",
+    "import requests\n",
+    "import pandas as pd\n",
+    "import pyarrow.parquet as pq\n",
+    "from io import BytesIO\n",
+    "import json\n",
+    "import yfinance as yf\n",
+    "from tqdm import tqdm\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "%env PG_CONNECTION_STRING=''  # nostra_mainnet.py has dependencies. To avoid error adding dummy env \n",
+    "import sys\n",
+    "sys.path.append('..')\n",
+    "from src.nostra_mainnet import ADDRESSES_TO_TOKENS as nostra_mainnet_tokens\n",
+    "from src.nostra_alpha import ADDRESSES_TO_TOKENS as nostra_alpha_tokens"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 168,
+   "id": "bd014fa2",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Processing zklend from Google Storage...\n",
+      "Processing nostra_alpha from Google Storage...\n",
+      "Processing nostra_mainnet from Google Storage...\n",
+      "Processing hashstack_v0 from Google Storage...\n",
+      "Processing hashstack_v1 from Google Storage...\n",
+      "Combined dataframe shape: (305033, 11)\n"
+     ]
+    }
+   ],
+   "source": [
+    "def load_data_from_google_storage(url):\n",
+    "    \"\"\"Load data from a Google Storage URL using pandas.read_parquet.\"\"\"\n",
+    "    response = requests.get(url)\n",
+    "    response.raise_for_status()  # Ensure the request was successful\n",
+    "\n",
+    "    with BytesIO(response.content) as f:\n",
```

Comment:
No need for `with BytesIO(response.content) as f:`, you can call `pandas.read_parquet` directly (use the path as a parameter).